### PR TITLE
[spring] Fixes #10878 - fixed dates in pathParam

### LINKED ISSUE #10878: [BUG][java] JavaSpring Date PathParams serialize in the wrong format in
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Date parameters in the path should be serialized according to RFC3339 (YYYY-MM-DD) but are actually serialized as M/D/YY. This was already addressed in the queryParams template by specifying a DateTimeFormat. The same logic should be used in the pathParams template.

##### openapi-generator version

Using OpenAPI Generator 5.3.0

##### OpenAPI declaration file content or url

```yaml
components:
  parameters:
    MyDateParam:
      name: myDate
      in: path
      required: true
      type: string
      format: date
```

##### Generation Details

Using the following settings:
 * module: JavaSpring
 * library: spring-cloud
 * dateLibrary: java8

##### Steps to reproduce

Serialize a pathParam date using a Feign client. The resulting path will contain an incorrect date.

##### Suggest a fix

The same issue was already addressed in the queryParams template. Applying it to the pathParams template resolves the issue. I'll submit a PR shortly.

### PR DESCRIPTION
**Pull request by jameswynn**
Fixes #10878 

This fixes how dates are serialized in JavaSpring's pathParam template by applying the same configuration as used by the queryParam template. With this change dates in the path will be serialized by Feign clients in correct RFC3339 notation (YYYY-MM-DD).

@cbornet

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by jameswynn** [comment]
#10878

File: modules/openapi-generator/src/main/resources/JavaSpring/pathParams.mustache [hunk 1]
```diff
@@ -1 +1 @@
...
+{{#isPathParam}}{{#useBeanValidation}}{{>beanValidationPathParams}}{{/useBeanValidation}}{{#oas3}}@Parameter(name = "{{baseName}}", description ={{/oas3}}{{^oas3}}@ApiParam(value ={{/oas3}} "{{{description}}}"{{#required}}, required = true{{/required}}{{#allowableValues}}, allowableValues = "{{#enumVars}}{{#lambdaRemoveDoubleQuote}}{{{value}}}{{/lambdaRemoveDoubleQuote}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/enumVars}}"{{/allowableValues}}{{^isContainer}}{{#defaultValue}}, defaultValue = "{{{.}}}"{{/defaultValue}}{{/isContainer}}){{#isDate}} @org.springframework.format.annotation.DateTimeFormat(iso = org.springframework.format.annotation.DateTimeFormat.ISO.DATE){{/isDate}} @PathVariable("{{baseName}}") {{>optionalDataType}} {{paramName}}{{/isPathParam}}
```
Comment (feech) [inline modules/openapi-generator/src/main/resources/JavaSpring/pathParams.mustache:1]: what about data-time?

Comment (jameswynn) [reply modules/openapi-generator/src/main/resources/JavaSpring/pathParams.mustache]: Good catch.

**Comment by jameswynn** [comment]
@bbdouglas @sreeshas @jfiala @lukoyanov @jeff9finger @karismann @Zomzog @lwlee2608

**Comment by wing328** [comment]
Closed via #11181. Please pull the latest master to give it a try.

**Pull request closed without merging**